src/new-components/teaser/teaser.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { theme } from '../../components/layout/theme';
+import { Link } from '../../components/links/links';
+import { IllustrationType } from '../../components/illustration/illustration-set';
+import {
+  Illustration,
+  IllustrationSize,
+} from '../../components/illustration/illustration';
+import { Icon } from '../icon/icon';
+import { TextStyles } from '../../components/typography/typography-v2';
+
+const TeaserContainer = styled.div<{ hover: boolean }>`
+  overflow: hidden;
+
+  ${(props) =>
+    props.hover &&
+    css`
+      color: ${theme.palette.text.topline};
+      cursor: pointer;
+    `};
+`;
+
+const StyledIllustration = styled(Illustration)<{ hover: boolean }>`
+  margin-bottom: 24px;
+  ${(props) =>
+    props.hover &&
+    css`
+      svg path {
+        fill: ${theme.palette.text.topline};
+      }
+    `}
+`;
+
+const ImageContainer = styled.div<{ hover: boolean }>`
+  overflow: hidden;
+  margin-bottom: 16px;
+
+  transition: transform 0.2s;
+  ${(props) => props.hover && 'transform: scale(1.05)'};
+`;
+
+const ToplineContainer = styled.div`
+  display: flex;
+  margin-bottom: 8px;
+  width: 100%;
+  justify-content: space-between;
+`;
+
+const Topline = styled.p`
+  ${TextStyles.toplineS};
+  color: ${theme.palette.text.topline};
+  margin: 0;
+`;
+
+const Timestamp = styled.p<{ hover: boolean }>`
+  ${TextStyles.timestamp};
+  color: ${(props) =>
+    props.hover ? theme.palette.text.topline : theme.palette.text.timestamp};
+  margin: 0;
+`;
+
+const TeaserText = styled.div`
+  ${TextStyles.textR};
+  margin: 0;
+`;
+
+const StyledTeaserTitle = styled.p<{
+  large: boolean;
+  hasTopline: boolean;
+}>`
+  ${TextStyles.headlineS}
+  ${(props) => props.large && TextStyles.headlineM};
+
+  margin-top: 0;
+  margin-bottom: ${(props) => (props.hasTopline ? '8px' : '16px')};
+`;
+
+const StyledIcon = styled(Icon)`
+  margin-top: 14px;
+`;
+
+export interface TeaserProps {
+  title: string;
+  topline?: string;
+  dateFormatted?: string | null;
+  teaserImage?: JSX.Element;
+  illustration?: IllustrationType;
+  linkTo?: string;
+  className?: string;
+  children: React.ReactNode | React.ReactNode[];
+}
+
+/**
+ * Teasers are used to link to another page and give the user a brief overview of the content of that page.
+ * This requires a headline, the path to the page (linkTo) and a short text, which is entered as a child.
```

Comment:
The bit about the "requires [...] path to the page" is not true anymore, right? We can have link-less teasers as far as I understand.
That means this comment is outdated and it's also in the wrong place. It documents `ConditionalLink` and I'm surprised that it is being picked up by Storybook. Looks more like a bug than a feature.
I suggest to add comments like this in the `mdx` file instead of the source. Let's take care of this when we introduce a proper mdx file.